ui/page/transaction/transactions_page.go

```diff
@@ -54,27 +51,45 @@ type TransactionsPage struct {
 
 	selectedTabIndex int
 
-	txTypeDropDown   *cryptomaterial.DropDown
+	txTypeDropDown *cryptomaterial.DropDown
+	walletDropDown *cryptomaterial.DropDown
+
 	transactionList  *cryptomaterial.ClickableList
 	previousTxFilter int32
 	scroll           *components.Scroll
 
-	tabs *cryptomaterial.ClickableList
+	tab *cryptomaterial.SegmentedControl
 
 	materialLoader material.LoaderStyle
+
+	sourceWalletSelector *components.WalletAndAccountSelector
+	assetWallets         []sharedW.Asset
+	selectedWallet       sharedW.Asset
+
+	isHomepageLayout,
+	showLoader,
+	showDisabledLayout bool
 }
 
-func NewTransactionsPage(l *load.Load) *TransactionsPage {
+func NewTransactionsPage(l *load.Load, isHomepageLayout bool) *TransactionsPage {
```

Comment:
This should take a `wallet` instead of taking `isHomepageLayout bool`. If a nil wallet is passed, then it's obvious that the page should show transactions for all wallets. But if a wallet is passed, the page should show transactions for that wallet. We don't care if the caller is the home page or some other page.

Reply:
Ermmmm... the wallet selection happens inside the Transactions page not the home page. Not sure how to get the wallet that will be passed to `NewtransactionsPage()`

Reply:
When the transactions page is opened from a specific wallet's home page. I think each specific wallet's home page is currently called MainPage.